src/features/DebugSession.ts

```diff
@@ -114,6 +124,87 @@ export class DebugSessionFeature extends LanguageClientConsumer
         ];
     }
 
+    public async createTemporaryIntegratedConsole(session: DebugSession) {
+        const settings = getSettings();
+        this.tempDebugProcess = await this.sessionManager.createDebugSessionProcess(settings);
+        this.tempSessionDetails = await this.tempDebugProcess.start(`DebugSession-${this.sessionCount++}`);
+
+        // NOTE: Dotnet attach debugging is only currently supported if a temporary console is used, otherwise we get lots of lock conflicts from loading the assemblies
+        if (session.configuration.attachDotnetDebugger) {
+            if (!vscode.extensions.getExtension("ms-dotnettools.csharp")) {
+                void this.logger.writeAndShowError("You specified attachDotnetDebugger in your PowerShell Launch configuration but the C# extension is not installed. Please install the C# extension and try again.");
+                return undefined;
+            }
+
+            // Will wait until the process is started and available before attaching
+            const pid = await this.tempDebugProcess.getPid();
+            if (pid === undefined) {
+                void this.logger.writeAndShowError("Attach Dotnet Debugger was specified but the PowerShell temporary debug session failed to start. This is probably a bug.");
+                return undefined;
+            }
+
+            const dotnetDebuggerConfig = this.getDotnetNamedConfigOrDefault(session, pid);
+            if (dotnetDebuggerConfig === undefined) {
+                void this.logger.writeAndShowError(`You specified dotnetDebuggerConfigName in your PowerShell Launch configuration but a matching launch config was not found. Please ensure you have a coreclr attach config with the name ${session.configuration.dotnetDebuggerConfigName} in your launch.json file or remove dotnetDebuggerConfigName from your PowerShell Launch configuration to use the defaults`);
+                return undefined;
+            }
+
+
+
+            // Ensure the .NET session stops before the PowerShell session so that the .NET debug session doesnt emit an error about the process unexpectedly terminating
```

Comment:
```suggestion
            // Ensure the .NET session stops before the PowerShell session so that the .NET debug session doesn't emit an error about the process unexpectedly terminating.
```